distribution/lib/Standard/Base/0.0.0-dev/src/Data/Statistics.enso

```diff
@@ -388,3 +344,78 @@ is_valid v = case v of
     Nothing -> False
     _ : Number -> v.is_nan.not
     _ -> True
+
+## PRIVATE
+type Accumulator
+    new : Integer -> Accumulator
+    new moments=0 = Accumulator.Value 0 Nothing Nothing (Vector.fill moments 0)
+
+    ## PRIVATE
+    Value count:Integer minimum:Any maximum:Any (moments:Vector|Nothing)
+
+    ## PRIVATE
+    comparator_error : Boolean
+    comparator_error self = self.minimum.is_nothing
+
+    ## PRIVATE
+    increment self value = if self.count == 0 then (Accumulator.Value 1 value value (self.update_moments value)) else
+        minimum = Accumulator.comparator self.minimum value Ordering.Less
+        maximum = Accumulator.comparator self.maximum value Ordering.Greater
+        Accumulator.Value self.count+1 minimum maximum (self.update_moments value)
+
+    ## PRIVATE
+    comparator current value bound = if current == Nothing then Nothing else
+        comp = Ordering.compare value current . catch Incomparable_Values (_-> Nothing)
+        if comp == Nothing then Nothing else
+            if comp == bound then value else current
+
+    ## PRIVATE
+    update_moments self value = if self.moments.is_nothing then Nothing else
+        case value of
+            _ : Number -> self.moments.map_with_index i->v-> v + (value ^ i+1)
+            _ -> Nothing
+
+    ## PRIVATE
+    compute : Statistic -> Any
+    compute self statistic =
+        case statistic of
+            Statistic.Count -> self.count
+            Statistic.Minimum -> if self.comparator_error then (Error.throw (Incomparable_Values.Error Nothing Nothing)) else self.minimum
+            Statistic.Maximum -> if self.comparator_error then (Error.throw (Incomparable_Values.Error Nothing Nothing)) else self.maximum
```

Comment:
If the comparison fails, I assume we are likely to still have access to which 2 values were not comparable. IMO it would be good to not lose this information, since it greatly helps with debugging (finding out which 2 values are incompatible). IMO we should try to almost never report `Incomparable_Values.Error Nothing Nothing` - we can figure out what were the offending values.

Reply:
It will now propagate the error from the compare method so we will get the correct value there now.

![image](https://github.com/enso-org/enso/assets/4699705/f6a556eb-7602-4c43-93cc-68d14aa38407)